services/offline/src/lib/online-status.ts

```diff
@@ -32,10 +41,19 @@ export function useOnlineStatus(
 
     // eslint-disable-next-line react-hooks/exhaustive-deps
     const updateState = useCallback(
-        debounce(
-            ({ type }: Event) => setOnline(type === 'online'),
-            options.debounceDelay || 1000
-        ),
+        debounce(({ type }: Event) => {
+            if (type === 'online') {
+                setOnline(true)
+            } else if (type === 'offline') {
+                setOnline(false)
+                if (!localStorage.getItem(lastOnlineKey)) {
```

Comment:
I don't think this logic is correct.

Take the following example:
1. User's local storage does not currently store last online and user is online
2. User then goes offline and last online is stored as `T`
3. User goes back online
4. At `T+N`, user goes offline but last online is still `T` as `!localStorage.getItem(lastOnlineKey)` is false.

Reply:
Ah, true - this needs to either remove the value from localStorage when going online, or check if `online` is true (i.e. `if (online || !localStorage.getItemlastOnlineKey) { ... }`)

Reply:
Opted for `online` check to reduce `localStorage` operations